dna_c_binding/src/lib.rs

```diff
@@ -58,10 +58,7 @@ pub extern "C" fn holochain_dna_to_json(ptr: *const Dna) -> *mut c_char {
             &*ptr
         };
 
-        let json = match dna.to_json() {
-            Ok(s) => s,
-            Err(_) => return std::ptr::null_mut(),
-        };
```

Comment:
This is the one case that I saw use `dna.to_json` and not call `unwrap`. In this case the behavior is still the same on failure though. It will panic and due to the `catch_unwind` it will still return `null_mut()`